vulnerabilities/importers/apache_tomcat.py

```diff
@@ -0,0 +1,150 @@
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with VulnerableCode or any VulnerableCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with VulnerableCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  VulnerableCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  VulnerableCode is a free software tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/vulnerablecode/ for support and download.
+
+import asyncio
+import dataclasses
+import re
+
+
+import requests
+from bs4 import BeautifulSoup
+from dephell_specifier import RangeSpecifier
+from packageurl import PackageURL
+
+from vulnerabilities.data_source import Advisory
+from vulnerabilities.data_source import DataSource
+from vulnerabilities.data_source import DataSourceConfiguration
+from vulnerabilities.data_source import Reference
+from vulnerabilities.helpers import create_etag
+from vulnerabilities.package_managers import MavenVersionAPI
+
+
+@dataclasses.dataclass
+class ApacheTomcatDataSourceConfiguration(DataSourceConfiguration):
+    etags: dict
+
+
+class ApacheTomcatDataSource(DataSource):
+
+    CONFIG_CLASS = ApacheTomcatDataSourceConfiguration
+    base_url = "https://tomcat.apache.org/security-{}"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.version_api = MavenVersionAPI()
+        asyncio.run(self.version_api.load_api({"org.apache.tomcat:tomcat"}))
+
+    def updated_advisories(self):
+        advisories = []
+        for advisory_page in self.fetch_pages():
+            advisories.extend(self.to_advisories(advisory_page))
+        return self.batch_advisories(advisories)
+
+    def fetch_pages(self):
+        tomcat_major_versions = {i[0] for i in self.version_api.get("org.apache.tomcat:tomcat")}
+        for version in tomcat_major_versions:
+            page_url = self.base_url.format(version)
+            if create_etag(self, page_url, "ETag"):
+                yield requests.get(page_url).content
+
+    def to_advisories(self, apache_tomcat_advisory_html):
+        advisories = []
+        page_soup = BeautifulSoup(apache_tomcat_advisory_html, features="lxml")
+        pageh3s = page_soup.find_all("h3")
+        vuln_headings = [i for i in pageh3s if "Fixed in Apache Tomcat" in i.text]
+        for data in vuln_headings:
+            fixed_version = data.text.split("Fixed in Apache Tomcat")[-1].strip()
```

Comment:
This may be more robust:
```
_left, _sep, fixed_version = data.text.lower().partition("fixed in apache tomcat")
fixed_version = fixed_version.strip()
```